service/lib/dinstaller/dbus/questions.rb

```diff
@@ -106,14 +86,40 @@ def all
       # @return [Array<DBus::Question>]
       attr_reader :exported_questions
 
+      # Builds the question path (e.g., org.opensuse.DInstaller/Questions1/1)
+      #
+      # @param question [DInstaller::Question]
+      # @return [::DBus::ObjectPath]
+      def path_for(question)
+        path = Pathname.new(PATH).join(question.id.to_s)
+
+        ::DBus::ObjectPath.new(path.to_s)
+      end
+
+      # Generates information about interfaces and properties for the given question
+      #
+      # Returns a hash containing interfaces names as keys. Each value is the same hash that would
+      # be returned by the org.freedesktop.DBus.Properties.GetAll() method for that combination of
+      # object path and interface. If an interface has no properties, the empty hash is returned.
+      #
+      # @param question [DBus::Question]
+      # @return [Hash]
+      def interfaces_and_properties(question)
+        get_all_method = self.class.make_method_name("org.freedesktop.DBus.Properties", :GetAll)
+
+        question.intfs.keys.each_with_object({}) do |interface, hash|
+          hash[interface] = question.send(get_all_method, interface).first
```

Comment:
I would just add here TODO to use public method when it will be available.

Reply:
In fact, it is a public method, so let's use `#public_send`.